domain/application/state/resource.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+
+	coreresource "github.com/juju/juju/core/resource"
+	"github.com/juju/juju/domain/application"
+	"github.com/juju/juju/internal/errors"
+)
+
+// buildResourceToAdd creates resources to add based on provided app and charm
+// resources.
+// Returns a slice of resourceToAdd and an error if any issues occur during
+// creation. Validates app resources against charm resources, warning if
+// mismatches are found.
```

Comment:
todo: update comment, there is no validation happening between app and charm resources here.